core/lib/dependency-graph/base-node.js

```diff
@@ -7,7 +7,8 @@
 /**
  * A union of all types derived from BaseNode, allowing type check discrimination
  * based on `node.type`. If a new node type is created, it should be added here.
- * @typedef {import('./cpu-node.js').CPUNode | import('./network-node.js').NetworkNode} Node
+ * @template [T=any]
+ * @typedef {import('./cpu-node.js').CPUNode<T> | import('./network-node.js').NetworkNode<T>} Node
```

Comment:
Assuming `CPUNode` does eventually need a generic, this type asserts that `T` will be the same between `NetworkNode` and `CPUNode` which would never be the case. Maybe we should just remove the generic from `CPUNode` for now?